apps/mull-ui/src/app/pages/profile/add-friends/add-friends.tsx

```diff
@@ -1,44 +1,152 @@
 import { faEllipsisH, faSearch, faUserPlus } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React, { useState } from 'react';
-import { mockUser1, mockUser2 } from '../../../../mockdata';
+import { History } from 'history';
+import { debounce } from 'lodash';
+import React, { useMemo, useState } from 'react';
+import { ROUTES } from '../../../../constants';
+import {
+  RelationshipType,
+  useAddFriendMutation,
+  useGetRelationshipsQuery,
+  useGetStrangersLazyQuery,
+  User,
+  useRemoveFriendMutation,
+} from '../../../../generated/graphql';
 import { CustomTextInput, MullBackButton } from '../../../components';
 import ContactRow from '../../../components/contact-row/contact-row';
 import './add-friends.scss';
-import SearchUsersPage from './search-users';
 
-export const AddFriendsPage = () => {
-  const [open, setOpen] = useState(false);
+export interface AddFriendsPageProps {
+  history: History;
+}
+
+export const AddFriendsPage = ({ history }: AddFriendsPageProps) => {
+  const [searchInput, setSearchInput] = useState<string>('');
+  const relData = useGetRelationshipsQuery();
+  const [getStrangers, strangerData] = useGetStrangersLazyQuery();
+  const [addFriend] = useAddFriendMutation();
+  const [removeFriend] = useRemoveFriendMutation();
+
+  const relMap = new Map<RelationshipType, User[]>();
+
+  // useMemo to avoid redefining debounce function every render.
+  const startSearch = useMemo(
+    () =>
+      debounce((searchInput: string) => {
+        getStrangers({ variables: { searchInput } });
+      }, 500),
+    [getStrangers]
+  );
+
+  const goToProfile = (id: string) => {
+    history.push(ROUTES.USER_BY_ID.replace(':id', id));
+  };
+
+  if (relData.loading || relData.error) {
+    return <div>something happened: {relData.error}</div>;
```

Comment:
so if it's loading it will display and error?

Reply:
Sorry about that, I kinda left this part intact after quickly writing something so I could continue working without errors.